src/client.ts

```diff
@@ -730,77 +727,62 @@ export class SpvWalletClient {
+   * @param {DraftTx} draftTransaction Draft transaction object
    * @return {string} Final transaction hex
-   */
-  SignTransaction(draftTransaction: DraftTransaction): string {
+   */  async SignTransaction(draftTransaction: DraftTx): Promise<string> {
     if (!this?.xPriv) {
       throw new ErrorNoXPrivToSignTransaction();
     }
 
-    const xPriv = this.xPriv as bsv.HDPrivateKey;
-    const txDraft: bsv.Transaction = new bsv.Transaction(draftTransaction.hex);
+    const xPriv = this.xPriv as HD;
+    const txDraft: Transaction = Transaction.fromHex(draftTransaction.hex);
 
-    // sign the inputs
-    const privateKeys: bsv.PrivateKey[] = [];
     draftTransaction.configuration.inputs?.forEach((input, index) => {
+      let hdWallet: HD | undefined;
       if (input.destination) {
         const dst = input.destination;
         // derive private key (m/chain/num)
-        let privKey = xPriv.deriveChild(dst.chain).deriveChild(dst.num);
+        hdWallet = xPriv.deriveChild(dst.chain).deriveChild(dst.num);
 
         if (dst.paymail_external_derivation_num != null) {
           // derive private key (m/chain/num/paymail_num)
-          privKey = privKey.deriveChild(dst.paymail_external_derivation_num);
+          hdWallet = hdWallet.deriveChild(dst.paymail_external_derivation_num);
         }
-
-        privateKeys.push(privKey.privateKey);
       }
 
       // small sanity check for the inputs
       if (
-        input.transaction_id != txDraft.inputs[index].prevTxId.toString('hex') ||
-        input.output_index != txDraft.inputs[index].outputIndex
+        input.transaction_id != txDraft.inputs[index].sourceTXID ||
+        input.output_index != txDraft.inputs[index].sourceOutputIndex
       ) {
         throw new ErrorTxIdsDontMatchToDraft(this.logger, input, index, txDraft.inputs[index]);
       }
 
-      // @todo add support for other types of transaction inputs
-      txDraft.inputs[index] = new bsv.Transaction.Input.PublicKeyHash({
-        prevTxId: input.transaction_id,
-        outputIndex: input.output_index,
-        script: new bsv.Script(input.script_pub_key),
-        output: new bsv.Transaction.Output({
-          script: new bsv.Script(input.script_pub_key),
-          satoshis: input.satoshis,
-        }),
-      });
+      if (hdWallet !== undefined) {
+        txDraft.inputs[index].unlockingScriptTemplate = new P2PKH().unlock(hdWallet.privKey, "single", false, input.satoshis, new P2PKH().lock(input.destination?.address as string));
+      }
+      txDraft.inputs[index].sourceOutputIndex = input.output_index;
+      txDraft.inputs[index].sourceTXID = input.transaction_id;
     });
 
-    txDraft.sign(privateKeys);
+    await txDraft.sign();
 
-    if (!txDraft.verify()) {
-      throw new ErrorDraftVerification(this.logger, txDraft);
-    }
-    if (!txDraft.isFullySigned()) {
-      throw new ErrorDraftFullySign(this.logger, txDraft);
-    }
-
-    return txDraft.toString();
+    return txDraft.toHex();
   }
 
+
```

Comment:
Unneeded empty line

Reply:
I think it was updated in your PR - can't find it